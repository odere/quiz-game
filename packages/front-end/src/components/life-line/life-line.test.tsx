import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import LifeLine from './life-line.view'

describe('components/LifeLine:', () => {
	it('should be able to show life states', () => {
		const lifeLineDefault = renderer
			.create(<LifeLine fiftyFifty plusTenSec />)
			.toJSON()
		expect(lifeLineDefault).toMatchSnapshot()
		const halfLife = renderer
			.create(<LifeLine fiftyFifty plusTenSec={false} />)
			.toJSON()
		expect(halfLife).toMatchSnapshot()
		const halfLifeInverted = renderer
			.create(<LifeLine fiftyFifty={false} plusTenSec />)
			.toJSON()
		expect(halfLifeInverted).toMatchSnapshot()
		const noLife = renderer
			.create(<LifeLine fiftyFifty={false} plusTenSec={false} />)
			.toJSON()
		expect(noLife).toMatchSnapshot()
	})
})
