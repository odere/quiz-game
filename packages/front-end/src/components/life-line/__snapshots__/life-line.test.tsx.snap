// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LifeLine: should be able to show life states 1`] = `
.c1 {
  background: #66bd38;
  color: #44168b;
  font-weight: 600;
  line-height: 1.5rem;
  padding: 0.25rem;
  text-align: center;
  -webkit-transition: 200ms;
  transition: 200ms;
  width: 50%;
}

.c1:hover {
  -webkit-filter: brightness(1.25);
  filter: brightness(1.25);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 1.5rem;
  gap: 0.25rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      onClick={[Function]}
    >
      50/50
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      onClick={[Function]}
    >
      +10s
    </div>
  </div>
</div>
`;

exports[`components/LifeLine: should be able to show life states 2`] = `
.c1 {
  background: #66bd38;
  color: #44168b;
  font-weight: 600;
  line-height: 1.5rem;
  padding: 0.25rem;
  text-align: center;
  -webkit-transition: 200ms;
  transition: 200ms;
  width: 50%;
}

.c1:hover {
  -webkit-filter: brightness(1.25);
  filter: brightness(1.25);
}

.c2 {
  background: #ff3859;
  color: #44168b;
  font-weight: 600;
  line-height: 1.5rem;
  padding: 0.25rem;
  text-align: center;
  -webkit-transition: 200ms;
  transition: 200ms;
  width: 50%;
}

.c2:hover {
  -webkit-filter: brightness(1.25);
  filter: brightness(1.25);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 1.5rem;
  gap: 0.25rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      onClick={[Function]}
    >
      50/50
    </div>
  </div>
  <div
    className="c2"
  >
    <div
      onClick={[Function]}
    >
      +10s
    </div>
  </div>
</div>
`;

exports[`components/LifeLine: should be able to show life states 3`] = `
.c2 {
  background: #66bd38;
  color: #44168b;
  font-weight: 600;
  line-height: 1.5rem;
  padding: 0.25rem;
  text-align: center;
  -webkit-transition: 200ms;
  transition: 200ms;
  width: 50%;
}

.c2:hover {
  -webkit-filter: brightness(1.25);
  filter: brightness(1.25);
}

.c1 {
  background: #ff3859;
  color: #44168b;
  font-weight: 600;
  line-height: 1.5rem;
  padding: 0.25rem;
  text-align: center;
  -webkit-transition: 200ms;
  transition: 200ms;
  width: 50%;
}

.c1:hover {
  -webkit-filter: brightness(1.25);
  filter: brightness(1.25);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 1.5rem;
  gap: 0.25rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      onClick={[Function]}
    >
      50/50
    </div>
  </div>
  <div
    className="c2"
  >
    <div
      onClick={[Function]}
    >
      +10s
    </div>
  </div>
</div>
`;

exports[`components/LifeLine: should be able to show life states 4`] = `
.c1 {
  background: #ff3859;
  color: #44168b;
  font-weight: 600;
  line-height: 1.5rem;
  padding: 0.25rem;
  text-align: center;
  -webkit-transition: 200ms;
  transition: 200ms;
  width: 50%;
}

.c1:hover {
  -webkit-filter: brightness(1.25);
  filter: brightness(1.25);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 1.5rem;
  gap: 0.25rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      onClick={[Function]}
    >
      50/50
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      onClick={[Function]}
    >
      +10s
    </div>
  </div>
</div>
`;
